src/js/containers/search/filters/awardID/AwardIDListContainer.jsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * AwardIDListContainer.jsx
+ * Created by michaelbray on 3/2/17.
+ */
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { isCancel } from 'axios';
+import _ from 'lodash';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as awardIDActions from 'redux/actions/search/awardIDActions';
+import * as AwardIDFormatter from 'helpers/awardIDFormatter';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+const propTypes = {
+    toggleAwardID: React.PropTypes.func,
+    setAutocompleteAwardIDs: React.PropTypes.func,
+    selectedAwardIDs: React.PropTypes.object,
+    piid: React.PropTypes.array,
+    fain: React.PropTypes.array,
+    uri: React.PropTypes.array,
+    parent_award__piid: React.PropTypes.array
+};
+
+export class AwardIDListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            awardIDSearchString: '',
+            autocompleteAwardIDs: [],
+            noResults: false
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.clearAutocompleteSuggestions = this.clearAutocompleteSuggestions.bind(this);
+        this.timeout = null;
+        this.fields = ['piid', 'fain', 'uri', 'parent_award__piid'];
+    }
+
+    componentDidMount() {
+        this.parseAutocompleteAwardIDs(this.props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(nextProps.piid, this.props.piid)
+            || !_.isEqual(nextProps.fain, this.props.fain)
+            || !_.isEqual(nextProps.uri, this.props.uri)
+            || !_.isEqual(nextProps.parent_award__piid, this.props.parent_award__piid)) {
+            this.parseAutocompleteAwardIDs(nextProps);
+        }
+    }
+
+    parseAutocompleteAwardIDs(awardIDProps) {
+        const values = [];
+        const fieldMap = {
+            piid: 'PIID',
+            fain: 'FAIN',
+            uri: 'URI',
+            parent_award__piid: 'PIID'
+        };
+
+        Object.keys(fieldMap).forEach((key) => {
+            const awardIDs = awardIDProps[key];
+            if (awardIDs && awardIDs.length > 0) {
+                awardIDs.forEach((item) => {
+                    if (item) {
+                        values.push({
+                            title: AwardIDFormatter.formatAwardID(item, fieldMap[key]),
+                            subtitle: fieldMap[key],
+                            data: Object.assign({}, item, {
```

Comment:
Do we need to retain the entire `item` or can we just grab the `id`? Especially as these objects get bigger and we're holding increasingly large amounts of data in state...